src/main/java/com/iridium/iridiumskyblock/gui/GUI.java

```diff
@@ -2,24 +2,48 @@
 
 import com.iridium.iridiumcore.utils.StringUtils;
 import com.iridium.iridiumskyblock.IridiumSkyblock;
+import com.iridium.iridiumskyblock.PlaceholderBuilder;
 import com.iridium.iridiumskyblock.configs.inventories.NoItemGUI;
-import lombok.AllArgsConstructor;
+import com.iridium.iridiumskyblock.database.Island;
 import lombok.NoArgsConstructor;
 import org.bukkit.Bukkit;
 import org.bukkit.event.inventory.InventoryClickEvent;
 import org.bukkit.inventory.Inventory;
 import org.bukkit.inventory.InventoryHolder;
 import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
 
 /**
  * Represents a clickable GUI.
  * Base for all other classes in this package.
  */
-@AllArgsConstructor
 @NoArgsConstructor
 public abstract class GUI implements InventoryHolder {
 
-    private NoItemGUI noItemGUI;
+    private @Nullable NoItemGUI noItemGUI;
+
+    /**
+     * The default constructor.
+     *
+     * @param noItemGUI The NoItemGUI of this GUI
+     * @param island    The island of this GUI. Can be null
+     */
+    public GUI(@NotNull NoItemGUI noItemGUI, @Nullable Island island) {
+        if (island != null) {
+            this.noItemGUI = new NoItemGUI(
```

Comment:
Not sure if this will work since the noItemGUI might not actually be a noItemGUI, everything just extends from a noItemGUI

Reply:
would it be better to just edit the getInventory method and do like List<Placeholder> placeholders = island==null?Collections.emptyList:new PlaceholderBuilder().applyIslandPlaceholders().build()

Reply:
Good point